The Big One

SageJFox opened the pull request:
I'm so sorry, little one.

SageJFox committed:
StayDuration updates

Added a client convar jazz_propfeed_stayduration to control how long entries in the prop feed/cash additions stay on screen without an update

Added fix so player money is shown immediately on spawn, rather than showing as $0 and getting an "addition" of their actual money count.

SageJFox committed:
Fix fix

Fixed the fix from commit 8bf9b910bf51771198dc55a6055a5a80f6c9dd70 that wound up preventing most map facts from being generated

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Reverted CalculateShardCount()

Reverted the changes to CalculateShardCount() in commit 70e31f0155010b5ae8fdf0c3e7921e1d943c75a3 with clarification on the label used. Utilizing area rather than length across without adjusting any other math made far, far too many shards spawn on what would amount to virtually any sized map (for reference, a map that was only the size of the platform that the level selector sits on in the jazz bar would generate 24 shards using area).

SageJFox committed:
Leaderboard Fix

Fixes "Props this Session" mistakenly being added twice when the prop vomiter starts.

SageJFox committed:
Idle dialog auto fix

Fixed idle dialog automatically dismissing as soon as it finished typing out, rather than letting the player click to continue.

SageJFox committed:
Fact Screen Fix Part II

Undoes the fix for commas after localization is parsed, so they properly display as commas again.

SageJFox committed:
merge

SageJFox committed:
Pianist Scene 3 update

Fixed Pianist being placed in the floor during the swiping scene.
Added the Good Shit.

SageJFox committed:
Bounds adjusting on cats

We now adjust the bounding box (for collisions, +use) for the cats depending on the sequence they're currently set to idle in.

SageJFox committed:
Editor Updates

Added some editor-only assets to assist in mapping work.

Updated "idleanim" property to "DefaultAnim" for jazz_cat so sequences set could be viewed in Hammer++ (entity currently takes both as a backup/so I don't have to push a half-finished map with this)

SageJFox committed:
Phys fixes

Fixed up phys meshes for some bar props

SageJFox committed:
Bartender Event 3

Added ability to ignite and extinguish actors to dialog commands. Totally unrelated to the Cellist I'm sure.

Fixed up scene setup, added sound to Cellist entering.

SageJFox committed:
Bar update - Mewseum NPC Start

jazz_bar:
- Put in areaportals, greatly reduces what needs to be rendered at once when in the main section of the bar.
- more updated details, nodraw shuffle
- Updated Zombie and Antlion displays in Mewseum, all displays given soundscapes.
- Started "NPC" pass in Mewseum
- more detail fixes

Fixed up jazzwalldetail, added version with ends capped.

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Textureloc location fix

Switched loading texture localization list from store init to gamemode init so it loads instead of errors from being "missing."

SageJFox committed:
Missed Localizations

Added missed localization in the intro and on the sign for the trolley

SageJFox committed:
Fixed mission logic

Who knew "not x == y" and "x ~= y" don't mean the same thing? Not I did!

SageJFox committed:
Missed Localization, II

Localized spawnmenu

Also fixed Jazztronauts weapons' icons not showing in the Jazztronauts spawn menu.

SageJFox committed:
CIRCLE UPDATE LET'S GO

Redid the distance alpha texture for the prop snatcher's border, should be much smoother now.

SageJFox committed:
whoops

Don't update something to not have an underscore so it's "easier to type it out" after you've already typed it out some, you'll forget to go back

SageJFox committed:
Weapon Descriptions

Added descriptions for weapons that didn't have them so GMod doesn't show an empty description when the Prop Snatcher is hovered.

Also minor fix to Bartender Event 5 to set the narrator at start.

SageJFox committed:
Feed message update

Death messages now much more accurately describe what killed the player and how

Feed background now better adjusts to size of text within it

Trolley names are now localized (and actually used)

Trolleys do DMG_VEHICLE in addition to DMG_CRUSH

SageJFox committed:
Mission update again

Added garbage composite props containing milk cartons/jugs to new Cellist 5 mission, as well as generalizing "cow". Should make mission much more doable.

SageJFox committed:
More scene fixes

Various fixes in various scenes. Typos, formatting, proper resets at scene cuts, fixing/adding sounds, grounding floating cats, you name it.

SageJFox committed:
Shard Message fix

Shard collected message properly displays price as affected by multiplier

SageJFox committed:
Added delay to clearing JazzDialogPVS

This gives us time to transition out of the scene (prevents dynamic props, etc. from immediately disappearing as soon as the player clicks the final time, before the transition has started)

SageJFox committed:
Sonador's model

Pushing this now mostly so it stops cluttering up my unstaged files list, it's not used (in a way that's pushed to the branch) just yet.

SageJFox committed:
weapon updates

switched run cleanup to default angles

enabled Hacker Goggles world model drawing

SageJFox committed:
More events, Cellist edition

Various event fixes. Lots of tabs and blank lines (I like consistent formatting okay), setting proper speakers, fixing animations/positions, adding some new animations, tweening, typos (yes those can exist in the Cellist's dialog)...

SageJFox committed:
bazz jar

Fixed prop chute trigger_hurt being disconnected from logic? Not sure when that happened.

Fixed up lighting origins for dark props in scene hallway

one last straggler from Cellist event fix

SageJFox committed:
cat_start update

:3

SageJFox committed:
Trolley out of grid fix

Adds a fix for the trolley attempting to spawn into or exit out of a wall that's so close to the max bounds of the map that "crazy physics" protection would kill it.

SageJFox committed:
Lua Error fix in voidfall

Fixed Lua client-side Error resulting from player spawning into the map directly onto a shard that then starts snatching brushes

SageJFox committed:
Flipped Stan and Run Bucket Positions

it was driving me crazy

SageJFox committed:
Singer Event fixes, part I: Flying cats edition

SageJFox committed:
fgd tweak

adds model parameter to fact screen

SageJFox committed:
jazz_door actually firing OnTeleport

SageJFox committed:
jazztable update

Fixed missing backfaces on portions of model, added LoD

SageJFox committed:
misc cleanup

Various fixes to singer event0, adds optional delay setting to fire dialog command

SageJFox committed:
> Consistency

Did a quick batch edit on all dialog so all scene descriptions start in a consistent way
> Like this, as opposed to
>Like this

SageJFox committed:
Killfeed update

Moved the Jazz-specific ones out of entity logic and into their own, so they can have fully custom death messages

SageJFox committed:
script macros

Added macros for script emotions, so it's more easily discernable what emotion cats are meant to display (hopefully especially helpful for translations, or if models ever get updated and the order changes)

Also inadvertently discovered and fixed a bunch of incorrect commands with this. "setskin" used where "setanim" was meant, and some typos in cat names that prevented them from being used (cat_bat, cat_car, cat_signer, all sorts of fun ones).

SageJFox committed:
Script Reversion

Went through the changes from Wertercatt with the knowledge that the program they used was probably shitting itself over the formatting stuff in these, un-Americanizing what got the Yankee treatment from it, and undoing proper grammar in dialog where it made sense for the character. Yes, the narrator is a character too.

Didn't fine-tooth comb the Thes because I know someone on the main branch already touched that up.

Also, some little fixes of my own sprinkled in.

SageJFox committed:
fgd update again

Couple more custom icons for the fgd, and some tweaks to the contents.

SageJFox committed:
Jazz_Intro tweaks

Updated credits scene
Fixed some small compile issues in various spots (wrong prop types, overlays with no faces, etc.)

Fixed missing "models/" path for snatching jamboree ending block gibs (no longer vomits errors)

Settled and slept all (or at least the vast majority of) physics props/ragdolls

Various little detail fixes

SageJFox committed:
whoops

Helps if the new texture for the credits was pushed too

SageJFox committed:
Merge Sanity I

First run though to attempt to get the merge to go more smoothly.

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

Dear God let's hope this is groovy.

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Pianist Event fixes

Minor touchups to Pianist events that I ran across while testing the merge.

SageJFox committed:
localization hotfix

So there's a bug with localization caching that we've run into because of the typo originally in the file's name. A blank copy of the old name needs to be included so the cache from it doesn't override the new file.

Reported this bug for GMod but who knows when/if it'd get fixed, and the localization was already put up on Steam so this would affect *everyone* who has downloaded that update and subsequently played Jazztronauts (or possibly just started up GMod, unclear on when they're cached).

Commenter25 commented:
So this is like... impossible to review?
![github literally saying infinite files were changed and a diff cannot be generated](https://github.com/Foohy/jazztronauts/assets/35016761/7d44030b-86ff-4ebe-96a2-d34c85ac0bdf)  
Even just clicking literally every commit, "[Bar update - Mewseum NPC Start](https://github.com/Foohy/jazztronauts/pull/166/commits/d3d51b2f0f10e68a300a2ece3a20f10196acb57b)" alone, and "[bazz jar](https://github.com/Foohy/jazztronauts/pull/166/commits/09ed8e01d796aea034cd76a3b7c9ceb76f8736b3)" alone, are unable to generate a diff.

I feel like this could've been split up to different branches over time? Less merge conflicts, people know what you've already done and don't conflict with it, easier to review and collaborate, things get merged faster... but I can't really criticize considering I was nearly planning on doing a similar thing. And at least it comes with a funny title :p

But hey, either way, from what I *can* tell, this looks fantastic! I see some things I had been thinking about trying to figure out, like areaportals in jazz_bar (it's always been so incredibly laggy for me). I'm also very very curious what "Updated credits scene" in jazz_intro means.

I don't *think* my PR for the intro conflicts with this, so I think I can leave it as it is, but I'm definitely gonna be changing what I had been working on for jazz_bar_intro... and I'm not sure how much I really need to mess with anything else, now that all the Grammarly Americanization and Correctification has been reverted (from what i can tell). Though I think I'll keep going through anyway, just for fun, it's been a learning experience and maybe it'll work out.

Also, interesting that the localization thing would still be an issue once it's on the workshop. It was really weird, it worked fine when the file name was the same, but then not if it was different, but then if I deleted the cached workshop file, it worked again. I just assumed it was some weird conflict with local installations and workshop installations.

SageJFox commented:
Any of them that include changes to a map are going to do that unfortunately. Any recompile of the bsp is gonna make the file look completely different to git and it's gonna give up trying to figure out what's changed in a 50mb binary file.

And yeah, we've been discussing that. We weren't expecting so many other people to be opening pull requests so suddenly so my branch was sorta just everything I've been touching up on and trying to get into a cohesive state before merging into the main. I usually work on a "if I see it broken, fix it if I can" basis so my stuff tends to be sorta scattershot in what it hits. It also tends to tie together (entity changes requiring map changes requiring asset changes) so splitting it into multiple branches isn't always as viable as it sounds on paper.

SageJFox commented:
Oh yeah, and on the last point, I reported what I thought was the issue for GMod itself, but in testing more today it does seem to be localization being taken from a disabled/unsubscribed workshop addon. If you wanted to add any of your observations to it, it's [over here](https://github.com/Facepunch/garrysmod-issues/issues/5466).

Commenter25 commented:
Yeah, I figure the maps will be impossible no matter what. It just makes it harder to look at everything else.

I honestly don't know where everyone else came from. I've had it set to notify me of activity in this repo for years (i forget why honestly), and I had noticed the big wertercatt PR after rarely seeing any activity here, and it started flooding back memories and made me wanna get some new friends I've met to play with me. And that gave me momentum to wanna help fix up this game now that I'm more technically knowledgeable, so it can hopefully be even better than it was before.

That's reasonable though, I think I kinda work the same way. I'm trying to contain it though, since there's a lot of benefits to keeping things separated.

Anyway, I downloaded this PR and played through the intro. I'm not sure what was changed other than the Error Explosion near the end, and the credits. In all honesty, I thought the errors were intentional, and its really funny that way. Since the whole of jazz_intro is kinda meant to be (as i understand) like you're playing through a really amateur and cheesy map, and it just kinda makes sense to be jank. Cats exploding out of it is a little funny too though, so I dunno.

SageJFox commented:
The cats were the intention, as far as things looked in the file. They were set to be used, but were lacking the "models/" at the beginning of the path, which is why the errors showed up. There's similar cases of other things in other spots, like sequences that were meant to be used but the name put in lacked an underscore, etc.

None of my changes for the intro (beyond the credits update) were meant to be big noticeable things, just fixing up what was already there to be functioning, or minor enhancements to make things smoother, like some player clipping I put in around the vent.

Commenter25 commented:
Yeah, I figured most of it wasn't meant to be noticeable. I do wonder if the credits could be polished up more eventually. Particularly, removing that flash of the player in the void at the start, adding a crash sound and shake when the bus leaves, and just smoothing out the timing and stutters. I have no idea how hard or even possible that is. I thought about investigating myself, but Hammer doesn't seem supported on Linux so I'd need to get a Windows setup available. And then I'd have to learn to understand Hammer along the way, so it'd just be a lot of hassle... so for now I'm sticking to the code, where I know I am cozy and probably not going to cause a catastrophe :p

SageJFox commented:
I was thinking of attempting that myself, yeah, but didn't get time for it. Same with getting the HUD disabled once the credits start. In due time I'm sure.

SageJFox committed:
The Localization never ends

Click to Continue...

SageJFox committed:
Explore Trolley Tweak

Updated so that a radio stolen from the Trolley doesn't reappear in the Void.

SageJFox committed:
jazz_hub convar

Added jazz_hub convar that allows setting a non-standard map for the hub.

SageJFox committed:
Spawn Menu Fix

Fixed non-store bought Jazztronauts weapons (Prop Snatcher, Portable Bus Stop, etc.) being unavailable in spawn menu without purchasing GMod spawn menu unlock.

SageJFox committed:
jazz_hub archive

yeah would probably help if it was actually remembered between map changes.

SageJFox committed:
Selector History Start

Adds ability for workshop browser to store and retrieve a history of addons viewed, as well as a simple edited jazz_bar that adds buttons to utilize this functionality.

SageJFox committed:
hub browser init update

Properly identifies if last visited map was local and sets the screen to it.

SageJFox committed:
info_busstop update, minor bar update

Updated info_busstop to no longer rely on offset from old sign position, instead uses its exact position

Updated fgd for this change, busstop now displays trolley model so its positioning can be exactly tracked.

Removed localization quickfix as it shouldn't be an issue once the workshop addon is updated.

jazz_bar updates:
- moved info_busstop in map to follow these changes (position adjusted slightly besides, trolley now goes down the direct middle of the tunnel)
- minor detailing consistency fixes between main map and scene areas
- fixed alignment on some map pictures, per issue #163
- seeing as I made it, moved crowbar sign from sunabouzu folder to mine

SageJFox committed:
hub selector output update

Updated selector with more outputs, OnMapDeselected and OnMapFailure (hopefully self-explanatory)

SageJFox committed:
One more for info_busstop

Updated so the spawned trolley's travel time and entrance/exit travel distances can be set by the info_busstop spawning it.

SageJFox committed:
PAC Lua Error fix

Fixes "ManipulateBoneScale" lua error when player is in dialog view while *setspeaker player* is used.

Our PAC implementation needs a lot of love but this at least calms the console spam down.

SageJFox committed:
...

localization

SageJFox committed:
bar5

harpoon fx

SageJFox committed:
PICK UP A WEEL ENDING

Fixed up some issues in the outro2 endings. Most noticeably, setting the speaker back after the player's first choice, and fixing up the branch names in the epilogue so they'd play out properly.

SageJFox committed:
selector/fgd update

Updated so selector turns towards player regardless of angle set on the entity (the old code required the selector be rotated to 0 270 0 to work properly)

Added "facing" property for the selector. Intended to control the axis (if any) that the rotation happens on, but right now it's either rotate as it is now or disable.

More fgd fixups: removed a lot of useless origins from stuff, and added the different options for specific facts to the fact screens, rather than having to type them out yourself.

SageJFox committed:
jazz_hub_minimum

Created an example vmf that contains just enough to make a functioning Jazztronauts hub, along with descriptions for why everything is needed.

SageJFox committed:
Factgen Time and Date localization

Converted from set format to sending over a laundry list of date/time info that the localization can then decide on using.

SageJFox committed:
prop vomiter marker

Updated prop vomiter to have its marker set in Hammer, rather than hardcoded to a specific point depending on the map.

SageJFox committed:
map updates (per vomit marker)

(Also threw in some super minor cleanups to some scripts I was poking at.)

jazz_bar unless specified (only jazz_bar compiled!)
- more detailing fixes
- r o t a t o
- touched up on some selector logic
- moved leaderboards per #172
- Updated prop vomiters on bar, outro, and hub_minimum with marker parameters (entire reason for this push)
- added stairs to outro/outro2 per #164

SageJFox committed:
SQL sanitation

Sanitizes map names for SQL (did you know that map names could contain apostrophes? I haven't seen that before today. Shoutouts to robot_pyro's_house.bsp for teaching me new Source horrors beyond my comprehension)

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Post-Merge fixup

Combed back through the intro after the merge to clean up some grossness.

SageJFox committed:
Removed redo

Can confirm that this file isn't used anywhere, and is just an older copy of the ending.

Just gonna cause potential confusion to continue having it.

SageJFox committed:
Oh my God found it finally

Removed `string string false` spam from console on selecting an addon, put in forever ago when I was testing something and forgot where it was.

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Auto Advance Tweak

Allows the setauto command to temporarily override the auto advance parameter.

SageJFox committed:
Merge remote-tracking branch 'Foohy/master'

SageJFox committed:
Memorial Merge Fixup II

Also fixes the need for the added text to the conditional (I think the text was fine, but I've watched this intro so many times already, and I'm sure there'll be many many more, and I don't want it to have to be any longer than it needs to be. x'd )

SageJFox committed:
jazz_dialog tweaks

Switched sendToAllPlayers flagged code to use GetHumans (left later bot check in, in case activator is a bot)

SageJFox committed:
Minor lua error fix

Fixes lua error if a mission turnin is played with jazz_debug_runscript

SageJFox committed:
Player Dialog consistency fix

Fixes inconsistency in dialog box model used between player dialog that's hard-scripted vs. player dialog picked from options.

Current behavior is that the player proxy is used if it is valid and the player doesn't currently have PAC enabled, otherwise the player themself is used.

If we can get the player proxy to play nicely with PAC, that part of the conditional can get removed, but right now there's too many possible break points with PAC.

SageJFox committed:
setcam improvement

Added optional fov setting on setcam (using this over setcam and fov separately prevents a noticeable hitch in the view between the two commands)

SageJFox committed:
setcam tweak

allows for "fov" label on new option so it doesn't get lost in the setang numbers

SageJFox committed:
Script teardown prep I

Updated spawn...() macros to set pose and skin to defaults.
Switched all scripts to using spawn...() macros.
Generally reformatted all scripts to be (much more) consistent.

Couple minor tweaks to some that got noticed while I was making sure I didn't fuck any of them up.

SageJFox committed:
setskin... fix?

Some dialog has setskin defined without a subject, just a skin index. So, added functionality for setskin in those situations to assume we want to set the skin on whoever is our current speaker.

SageJFox committed:
Script Teardown Prep II

Throwing the missions in now that I got to properly test them.

SageJFox committed:
Teardown Prep III

Mostly just finished up merging fov settings into camera settings that're meant to happen together.

SageJFox committed:
Workshop Download Fix

Fixes issue where a workshop id of nil or 0 on the browser locks up the level selector if it's selected (usually happens after launching server on a local map and then returning to the hub)

Also a fix so fact screen shows listed sub-error for download failures (not bothering to localize these at the moment, but we *could*).

SageJFox committed:
DOOOOOOH MY FOOH-

Dude, the thing that everybody-

SageJFox committed:
Jazz Portal update

Fixed up weighting and tweaked animations on Jazzportal.mdl

Adjusted "close" sequence so the root stays in the basin on return, and pull goes up to the magnifying glass.

SageJFox committed:
factgen tweak

Fixes comments with commas in the Steam account name

SageJFox committed:
jazzwallpaper fixes

removed duplicate of jazzwallpaper01_a.vtf that 02 was using

Fixed envmap settings on 03

Added keywords to 02 and 03

SageJFox committed:
Jazz Ivy LoDs

Added/improved LoD models for ivy props.

Fixed smoothing on leaves for 03-08

SageJFox committed:
Ivy miss

Huh, missed this tweak on the QC in the upload last night.

SageJFox committed:
Minor ChatFade fix

Switches chat fade from using frametime to the difference bewteen curtime and the last time it was updated. This fixes the chat window "sticking" if the cat in question was removed from the PVS/otherwise not rendering while a fade was happening (such as the player going through an areaportal while not looking at the cat.)

SageJFox committed:
Disable Auto-Switching off Jazz weapons

Prevents picking up a crossbow, .357, etc. from yoinking a player out of the Jazz weapons. Most for functional reasons (don't want to interrupt the player snatching if they happen to pick up a weapon, or if they're trying to call the bus, or bore with Stan, or not die to fall damage from Run, etc.)

SageJFox committed:
Respawn localization

Also adds proper key bind checking

SageJFox committed:
Localization fixup

Switched `dmg.` tokens to `jazz.dmg.` to be less likely to conflict with other addons.

Changed npc/entity/weapon tokens to include "jazz." at the beginning localization, as otherwise they override default values outside of Jazztronauts, too. To support this:

Updated functionality of jazzloc.localize() to look for a Jazztronauts version of a default localization before falling back on the default.

Also also, adds support for renaming keys (i.e. showing "Right click" for "MOUSE2")

SageJFox committed:
Static Props retain set skin when snatched

SageJFox committed:
misc touchups

been sitting in stashes for *too long*

Bunch of entities for accepted snatching classes, fixes "IgnoreForSnatch" to disallow snatching.

Fixes bus beam to draw over translucent objects (doesn't draw over the static prop proxies, because those are done in a 2D context???).

SageJFox committed:
Merge branch 'master' into SelectorHistory

SageJFox committed:
Merge prep

Decided against doing previews for items in history (at this time), while they'd be nice they aren't necessary for this to be added, the functionality itself is more pressing.

SageJFox committed:
Merge branch 'SelectorHistory'

SageJFox committed:
Credits Update

c:

SageJFox committed:
You, the person reading this right now, should use the Sceneroots branch

fixed old host on main branch. For most up to date work on this branch, see https://github.com/SageJFox/jazztronauts/tree/sceneroots
